packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -209,52 +224,221 @@ describe("Garbage Collection Tests", () => {
+                validateSweepTimeout();
+
+                const expectedMetadata: IGCMetadata = { sweepEnabled: false, gcFeature: 1, sessionExpiryTimeoutMs: defaultSessionExpiryDurationMs };
+                const outputMetadata = gc.getMetadata();
+                assert.deepEqual(outputMetadata, expectedMetadata, "getMetadata returned different metadata than expected");
+            }
+            it("TestOverride.SessionExpiryMs setting applied to timeout but not written to file - New Container", () => {
+                injectedSettings[testOverrideSessionExpiryMsKey] = 789;
+                gc = createGcWithPrivateMembers();
+                testSessionExpiryMsOverride();
+            });
+            it("TestOverride.SessionExpiryMs setting applied to timeout but not written to file - Existing Container", () => {
+                injectedSettings[testOverrideSessionExpiryMsKey] = 789;
+                gc = createGcWithPrivateMembers({ sessionExpiryTimeoutMs: defaultSessionExpiryDurationMs, gcFeature: 1 } /* metadata */);
+                testSessionExpiryMsOverride();
+            });
+            it("RunSessionExpiry setting turned off", () => {
+                injectedSettings[runSessionExpiryKey] = false;
+                injectedSettings[testOverrideSessionExpiryMsKey] = 1234; // This override should be ignored
+                gc = createGcWithPrivateMembers();
+                assert.equal(gc.sessionExpiryTimeoutMs, undefined, "sessionExpiryTimeoutMs should be undefined if runSessionExpiryKey setting is false");
+                assert.equal(gc.sessionExpiryTimer, undefined, "sessionExpiryTimer should be undefined if it's disabled");
+                validateSweepTimeout();
+            });
+            it("Disable SessionExpiry", () => {
+                injectedSettings[disableSessionExpiryKey] = true;
+                injectedSettings[testOverrideSessionExpiryMsKey] = 1234; // This override should be ignored
+                gc = createGcWithPrivateMembers();
+                assert.equal(gc.sessionExpiryTimeoutMs, defaultSessionExpiryDurationMs, "sessionExpiryTimeoutMs should be set even if disabled");
+                assert.equal(gc.sessionExpiryTimer, undefined, "sessionExpiryTimer should be undefined if it's disabled");
+                // (validateSweepTimeout doesn't try to handle this case)
+                assert.equal(gc.sweepTimeoutMs, undefined, "sweepTimeoutMs should be undefined if sessionExpiryTimeoutMs is undefined");
+            });
         });
 
+        describe("Session Behavior (e.g. 'shouldRun' fields)", () => {
+            beforeEach(() => {
+                injectedSettings["Fluid.GarbageCollection.TestOverride.InactiveTimeoutMs"] = 1; // To ensure it's less than sweep timeout
+            });
+
+            describe("shouldRunGC", () => {
+                const testCases: [{ gcEnabled: boolean; disableGC?: boolean; runGC?: boolean; }, boolean /* expectedResult */][] = [
```

Comment:
I would put this as one object, 
`[{ gcEnabled: boolean; disableGC?: boolean; runGC?: boolean; expectedShouldRunGC: boolean}]`

Reply:
I started that way, but switched to keep the expected value out of the test name (see line 368)